src/poggit/ci/builder/DefaultProjectBuilder.php

```diff
@@ -172,6 +174,112 @@ protected function build(Phar $phar, RepoZipball $zipball, WebhookProjectModel $
+        $id = "phpstan-".(Meta::getRequestId() ?? bin2hex(random_bytes(8)));
+
+        Meta::getLog()->v("Starting PHPStan flow with ID '{$id}'");
+
+        Lang::myShellExec("docker create --name {$id} jaxkdev/poggit-phpstan:0.0.5", $stdout, $stderr, $exitCode);
+
+        if($exitCode !== 0){
+            $status = new PhpstanInternalError();
+            $status->exception = "PHPStan failed with ID '{$id}', contact support with the ID for help if this persists.";
+            $result->addStatus($status);
+            Meta::getLog()->e("Failed to create docker container with id '{$id}', Status: {$exitCode}, stderr: {$stderr}");
+            return;
+        }
+
+        Meta::getLog()->v("Copying plugin '{$phar}' into '{$id}:/source/plugin.phar'");
+
+        Lang::myShellExec("docker cp {$phar} {$id}:/source/plugin.phar", $stdout, $stderr, $exitCode);
+
+        if($exitCode !== 0){
+            $status = new PhpstanInternalError();
+            $status->exception = "PHPStan failed with ID '{$id}', contact support with the ID for help if this persists.";
+            $result->addStatus($status);
+            Meta::getLog()->e("Failed to copy '{$phar}' into '{$id}:/source/plugin.phar', Status: {$exitCode}, stderr: {$stderr}");
+            return;
+        }
+
+        Meta::getLog()->v("Starting container '{$id}'");
+
+        Lang::myShellExec("docker start -ia {$id}", $stdout, $stderr, $exitCode);
+
+        if($exitCode !== 0 and $stderr === ""){ //Exits with 1 if problems found...
+            $status = new PhpstanInternalError();
+            $status->exception = "PHPStan failed with ID '{$id}', contact support with the ID for help if this persists.";
+            $result->addStatus($status);
+            Meta::getLog()->e("Failed to start container '{$id}', Status: {$exitCode}, stderr: {$stderr}");
+            return;
+        }
+
+        if($exitCode === 255 and substr($stderr,0,11) === "Parse error"){
+            $status = new PhpstanLint();
+            $status->message = str_replace("/source/", "", $stderr);
+            $result->addStatus($status);
+            return;
+        }
+
+        $tmpFile = Meta::getTmpFile(".json"); //Extension is f*****
+
+        Meta::getLog()->v("Copying results from container '{$id}' into temp file '{$tmpFile}'");
+
+        Lang::myShellExec("docker cp {$id}:/source/phpstan-results.json {$tmpFile}", $stdout, $stderr, $exitCode);
+
+        if($exitCode !== 0){
+            $status = new PhpstanInternalError();
+            $status->exception = "PHPStan failed with ID '{$id}', contact support with the ID for help if this persists.";
+            $result->addStatus($status);
+            Meta::getLog()->e("Failed to copy results from container '{$id}', Status: {$exitCode}, stderr: {$stderr}");
+            return;
+        }
+
+        $data = json_decode(file_get_contents($tmpFile), true);
+
+        //@unlink($tmpFile); TODO uncomment, debug purposes.
```

Comment:
Consider using `if(Meta::isDebug())` here

Reply:
Majority of code is based off me debugging, none of this is ready for production

Reply:
Sorted.